cmd/feegrant.go

```diff
@@ -0,0 +1,188 @@
+			} else if prov.PCfg.FeeGrants != nil && granterKey != prov.PCfg.FeeGrants.GranterKey && update {
+				prov.PCfg.FeeGrants.GranterKey = granterKey
+				cfgErr := a.OverwriteConfig(a.Config)
+				cobra.CheckErr(cfgErr)
+			}
+
+			if prov.PCfg.FeeGrants == nil || updateGrantees || len(grantees) > 0 {
+				var feegrantErr error
+
+				//No list of grantees was provided, so we will use the default naming convention "grantee1, ... granteeN"
+				if grantees == nil {
+					feegrantErr = prov.ConfigureFeegrants(numGrantees, granterKey)
+				} else {
+					feegrantErr = prov.ConfigureWithGrantees(grantees, granterKey)
+				}
+
+				if feegrantErr != nil {
+					return feegrantErr
+				}
+
+				if feegrantErr != nil {
+					return feegrantErr
+				}
+
+				cfgErr := a.OverwriteConfig(a.Config)
+				cobra.CheckErr(cfgErr)
+			}
+
+			memo, err := cmd.Flags().GetString(flagMemo)
+			if err != nil {
+				return err
+			}
+
+			ctx := cmd.Context()
+			_, err = prov.EnsureBasicGrants(ctx, memo)
+			if err != nil {
+				return fmt.Errorf("error writing grants on chain: '%s'", err.Error())
+			}
+
+			//Get latest height from the chain, mark feegrant configuration as verified up to that height.
+			//This means we've verified feegranting is enabled on-chain and TXs can be sent with a feegranter.
+			if prov.PCfg.FeeGrants != nil {
+				fmt.Printf("Querying latest chain height to mark FeeGrant height... \n")
+				h, err := prov.QueryLatestHeight(ctx)
+				cobra.CheckErr(err)
+				prov.PCfg.FeeGrants.BlockHeightVerified = h
+				fmt.Printf("Feegrant chain height marked: %d\n", h)
+				cfgErr := a.OverwriteConfig(a.Config)
+				cobra.CheckErr(cfgErr)
+			}
+
+			return nil
+		},
+	}
+	cmd.Flags().BoolVar(&update, "overwrite-granter", false, "allow overwriting the existing granter")
+	cmd.Flags().BoolVar(&updateGrantees, "overwrite-grantees", false, "allow overwriting existing grantees")
+	cmd.Flags().IntVar(&numGrantees, "num-grantees", 10, "number of grantees that will be feegranted with basic allowances")
+	cmd.Flags().StringSliceVar(&grantees, "grantees", []string{}, "comma separated list of grantee key names (keys are created if they do not exist)")
+	cmd.MarkFlagsMutuallyExclusive("num-grantees", "grantees")
+
+	memoFlag(a.Viper, cmd)
+	return cmd
+}
+
+func feegrantBasicGrantsCmd(a *appState) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "basic chain-name [granter]",
+		Short: "query the grants for an account (if none is specified, the default account is returned)",
+		Args:  cobra.RangeArgs(1, 2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			chain := args[0]
+			cosmosChain, ok := a.Config.Chains[chain]
+			if !ok {
+				return errChainNotFound(args[0])
+			}
+
+			prov, ok := cosmosChain.ChainProvider.(*cosmos.CosmosProvider)
+			if !ok {
+				return errors.New("only CosmosProvider can be feegranted")
+			}
+
+			// TODO fix pagination
+			// pageReq, err := client.ReadPageRequest(cmd.Flags())
```

Comment:
are these TODOs still relevant?

Reply:
I think so. Most other funcs in the relayer have the same TODOs/issue. For good default behavior its probably OK, but it should be fixed.